src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -96,72 +144,174 @@ export const MultiSelect: FC<MultiSelectProps> = ({
         onSelectionChange(Array.from(keySet));
     };
 
-    const summarizedLabel = [activeItemKeys.length, `option${activeItemKeys.length > 1 ? 's' : ''}`, 'selected'].join(
-        ' ',
-    );
+    const handleSpacebarToggle = (e: KeyboardEvent<HTMLDivElement>) => {
+        if (e.code === 'Space' && (filterInputRef.current !== document.activeElement || !filterable)) {
+            toggleOpen();
+        }
+    };
+
+    const getDecoratorClasses = () => {
+        if (emphasis === MultiSelectEmphasis.Weak) {
+            return hasSelectedItems ? '' : 'tw-text-text-weak';
+        } else {
+            return '';
+        }
+    };
+
+    const getTagType = () => {
+        if (emphasis === MultiSelectEmphasis.Weak) {
+            return TagType.Selected;
+        } else {
+            if (open) {
+                return TagType.SelectedWithFocus;
+            } else {
+                return TagType.Selected;
+            }
+        }
```

Comment:
I believe you can simplify this logic, WDYT?

```suggestion
        if (emphasis === MultiSelectEmphasis.Weak || !open) {
            return TagType.Selected;
        }

        return TagType.SelectedWithFocus;
```